src/terrain/draw_terrain_raster.js

```diff
@@ -260,7 +262,9 @@ function drawTerrainForGlobe(painter: Painter, terrain: Terrain, sourceCache: So
 
 function drawTerrainRaster(painter: Painter, terrain: Terrain, sourceCache: SourceCache, tileIDs: Array<OverscaledTileID>, now: number) {
     if (painter.transform.projection.name === 'globe') {
+        painter.gpuTimingGlobeStart();
```

Comment:
One noteworthy thing is that this might be called multiple times for each tile depending on the style. In deferred/terrain mode we're rendering drapeable layers to offscreen textures in batches before accumulating them to the screen. This behavior would need to be checked before collecting any measures

Reply:
Do you mean that it may be called before `painter.gpuTimingGlobeEnd` is called? Also note that this is the only place that this function is invoked. I was using the same style to compare before and after cases so I am not sure how this could impact the measurements which would be identical in both cases except that with this change we have fewer vertices. Could you please elaborate on this? @mpulkki-mapbox

Reply:
You should check how `drawTerrainRaster` is invoked. I believe it can be called multiple times per frame. Is the timer correctly accumulating total frame time or just capturing the last call?

Reply:
Two things about timer queries:
1. Calling begin/end pairs consequentially doesn't reset the timer
2. The elapsed time is not cumulative
so for example:

the gpu time gathered with this snippet 
```
ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, query);
for (let i = 0; i < 1000; i++) {
  draw();
}
ext.endQueryEXT(ext.TIME_ELAPSED_EXT);
``` 

is not the same as the timing gathered this way:
```
for (let i = 0; i < 1000; i++) {
  ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, query);
  draw();
  ext.endQueryEXT(ext.TIME_ELAPSED_EXT);
}
``` 
I still think the the measurement comparisons here are valid but I think we could additionally add a counter which indicates how many times per frame we draw the globe tiles. When reading the gpu time we then multiply it by this counter to acquire the total gpu time. @mpulkki-mapbox what do you think?

Reply:
you could accumulate the elapsed time manually to some variable? I also liked the idea from @karimnaaji that this could be moved one level up to measure all deferred rendering. Could this be moved all the way to painter.js?

Reply:
The result of a timer query isn't immediately available. Currently we wait 50ms for the results to become available. Moving the code levels higher would solve this issue but would not be well scoped anymore. I don't have any solution of how we could accumulate results without using multiple queries with the number being dependent on the number of times this piece of code gets executed. This is highly convoluted and impractical IMHO.

Reply:
@akoylasar @mpulkki-mapbox I went ahead and added the suggestion I proposed, as like @mpulkki-mapbox mentioned I don't think the numbers would be accurate since we can have several dispatch to main framebuffer after reaching the max framebuffer pool in the terrain rendering. This also looked suspiciously low for a full deferred render over the screen.

Here a the new results, which are also promising!

Before this PR:
```
Meauring finished!
Average globe gpu render time is 2.8423135402408968(ms)
```
After this PR:
```
Meauring finished!
Average globe gpu render time is 1.5943233486112875(ms)
```
Please take a look at https://github.com/mapbox/mapbox-gl-js/pull/11638 so we can move this PR forward.

Reply:
@karimnaaji Awesomeness! Thanks a lot Karim I was unfortunately busy with something else and wanted to make sure we have a unanimous decision before continuing with this PR. The change you've added is great I've left one comment only for clarification. Let's merge this! :)

Reply:
So I ran the measurements once again and the numbers I on my machine (with a `Nvidia GeForce RTX 2080 Max-Q` on `Ubuntu 20.0.4.3 LTS`) are as follows:

before:
```
Meauring finished!
Average globe gpu render time is 0.37357575409578475(ms)
```

after:
```
Meauring finished!
Average globe gpu render time is 0.338998169518658(ms)
```
Note: To switch to "before" case I simply return 0 from `getLatitudinalLod` in `globe_util.js`. @karimnaaji were you running this same measurement (in `flytolocations.html`)? And was it on mobile or desktop? The readings in my case might be due to the fact that on this high-end GPU the changes don't make much of a difference.

Reply:
>@karimnaaji were you running this same measurement (in flytolocations.html)?

Yes!

>And was it on mobile or desktop? 

I tested on a macbook 2019.

>The readings in my case might be due to the fact that on this high-end GPU the changes don't make much of a difference.

Yes I would think so, that's a top notch GPU :) so I wouldn't be surprised of having little difference like this.